Update Setup Guide to include Cert Management for Garden (#12706)
gardener/gardener: docs/deployment/setup_gardener.md

- [Gardener Configuration](../operations/configuration.md#system-configuration)
- [DNS records](../extensions/resources/dnsrecord.md)

### Certificates

> Some components of a Gardener landscape require TLS certificates signed by well-known, trusted certificate authority (CA). While it is certainly possible to provide certificates manually, using an ACME service is much more convenient and recommended. Utilizing the [Gardener extension for certificate services](https://github.com/gardener/gardener-extension-shoot-cert-service) allows to automate the creation and renewal of certificates for the Gardener landscape.

Reviewer: ```suggestion
Some components of a Gardener landscape require TLS certificates signed by a well-known, trusted certificate authority (CA). While it is certainly possible to provide certificates manually, using an ACME service is much more convenient and recommended. Utilizing the [Gardener extension for certificate services](https://github.com/gardener/gardener-extension-shoot-cert-service) allows automating the creation and renewal of certificates for the Gardener landscape.
```

### Backup Bucket

The virtual Garden cluster's `etcd` is governed by [etcd-druid](https://github.com/gardener/etcd-druid). If configured, backups are created continuously and stored to a supported object store. This requires credentials to manage an object store on the chosen infrastructure.
It is recommended for any productive installation of Gardener.

### Infrastructure

> For each infrastructure supported by Gardener a so-called __provider-extension__ exists. It implements the required interfaces. For this example, the [OpenStack extension](https://github.com/gardener/gardener-extension-provider-openstack) will be used. Of course, a single Gardener landscape can support multiple infrastructures, which is a common use case. In this case, the respective provider-extensions need to be deployed as well.

Reviewer: ```suggestion
For each infrastructure supported by Gardener, a so-called __provider-extension__ exists. It implements the required interfaces. For this example, the [OpenStack extension](https://github.com/gardener/gardener-extension-provider-openstack) will be used. Of course, a single Gardener landscape can support multiple infrastructures, which is a common use case. In this case, the respective provider-extensions need to be deployed as well.
```

When setting up a new Gardener landscape, choose the latest version and either install the chart or render the templates e.g.:

```bash
helm template ./charts/gardener/operator --namespace garden --set replicaCount=2 --set image.tag=v1.125.0
```

Reference documentation:

- [Gardener Operator](../concepts/operator.md)

### Extensions

Gardener is provider-agnostic and relies on a growing set of extensions. An extension can provide support for an infrastructure (required) or add a desired feature to a Kubernetes cluster (often optional).

There are a few extensions considered essential to any Gardener installation:

- at least one [infrastructure provider](../../extensions/README.md#infrastructure-provider) (commonly referred to as provider-extension)
- at least one [operating system extension](../../extensions/README.md#operating-system) - [Garden Linux](https://github.com/gardener/gardener-extension-os-gardenlinux) is recommended
- at least one [network plugin (CNI)](../../extensions/README.md#network-plugin)
- Shoot services

> An extension commonly consists of two components - a controller implementing the Gardener's extension contract and an admission controller. The latter is deployed to the runtime cluster and prevents misconfiguration of `Shoot`s. While an extensions primary purpose is to implement functionality for `Shoot` clusters, it can also be deployed to the runtime cluster to provide additional functionality like managing `DNSRecord`s or `BackupBucket`s using the well-known and proven code paths.

Reviewer: ```suggestion
An extension commonly consists of two components – a controller implementing the Gardener's extension contract and an admission controller. The latter is deployed to the runtime cluster and prevents misconfiguration of `Shoot`s. While an extension's primary purpose is to implement functionality for `Shoot` clusters, it can also be deployed to the runtime cluster to provide additional functionality like managing `DNSRecord`s or `BackupBucket`s using the well-known and proven code paths.
```

#### Choices

For this example, the following choices are made:
- [Garden Linux OS extension](https://github.com/gardener/gardener-extension-os-gardenlinux)
- [Provider OpenStack](https://github.com/gardener/gardener-extension-provider-openstack)
- [Calico](https://github.com/gardener/gardener-extension-networking-calico)
- [DNS service](https://github.com/gardener/gardener-extension-shoot-dns-service)
- [Cert service](https://github.com/gardener/gardener-extension-shoot-cert-service)
- [OIDC service](https://github.com/gardener/gardener-extension-shoot-oidc-service)

Reference documentation:
- [Extension Contract](../extensions/resources/extension.md#contract-extension-resource)
- [GEP 01 Extensibility](../proposals/01-extensibility.md)
- [Extension Library](../../extensions/README.md#known-extension-implementations)

#### How to install an extension

> An extension is commonly described by an `Extension` resources (API group `operator.gardener.cloud/v1alpha1`) in the runtime cluster. The Gardener Operator will take care of deploying the required resources both to the runtime and to the virtual Garden cluster. More details can be found in the [Extension Resource documentation](../concepts/operator.md#extension-resource).

Reviewer: ```suggestion
An extension is commonly described by an `Extension` resource (API group `operator.gardener.cloud/v1alpha1`) in the runtime cluster. The Gardener Operator will take care of deploying the required resources both to the runtime and to the virtual Garden cluster. More details can be found in the [Extension Resource documentation](../concepts/operator.md#extension-resource).
```

> Typically, only a few extensions are needed on the runtime cluster and the Gardener Operator manages them directly through `ManagedResources`. Most of the extensions however, are registered to the virtual Garden cluster and then installed onto the `Seeds`.

Reviewer: ```suggestion
Typically, only a few extensions are needed on the runtime cluster, and the Gardener Operator manages them directly through `ManagedResource`s. Most of the extensions, however, are registered to the virtual Garden cluster and then installed onto the `Seed`s.
```

> In order to make an extension known to the `Garden`, the Gardener Operator translates the `Extension` into two resources and applies them to the virtual Garden cluster - firstly, a `ControllerDeployment` and secondly, a matching `ControllerRegistration`. With both in place, Gardener-managed `ControllerInstallation`s take care of the actual deployment of the extension to the target environment.

Reviewer: ```suggestion
To make an extension known to the `Garden`, the Gardener Operator translates the `Extension` into two resources and applies them to the virtual Garden cluster – firstly, a `ControllerDeployment` and secondly, a matching `ControllerRegistration`. With both in place, Gardener-managed `ControllerInstallation`s take care of the actual deployment of the extension to the target environment.
```

Check the [extension registration documentation](../extensions/registration.md) for more details.

> The `ControllerRegistration` is also a useful resource to check, as it declares, which resources it implements. An example of such a `ControllerRegistration`, can be found in the [provider OpenStack](https://github.com/gardener/gardener-extension-provider-openstack) extension's [example file](https://github.com/gardener/gardener-extension-provider-openstack/blob/master/example/controller-registration.yaml).

Reviewer: ```suggestion
The `ControllerRegistration` is also a useful resource to check, as it declares which resources it implements. An example of such a `ControllerRegistration` can be found in the [provider OpenStack](https://github.com/gardener/gardener-extension-provider-openstack) extension's [example file](https://github.com/gardener/gardener-extension-provider-openstack/blob/master/example/controller-registration.yaml).
```
